### PR TITLE
Neo7337/feature835

### PR DESCRIPTION
**Pull request by neo7337**
Raising this PR for the feature reqeust for #835 
The data has been filtered on the base 30 days but is configurable

### CODE REVIEW

File: utils/cache.js [hunk 1]
```diff
@@ -12,7 +12,27 @@ exports.currentStatus = {
...
+/**
+ * Parses NYT Counties data
+ * @param 	{string}	lastdays  	How many days to show always take lastest
+ * @returns {Array}				The filtered historical data.
+ */
+exports.nytCounties = (lastdays = 30) => {
+	if (lastdays === 'all') {
+		return this.currentStatus.nytCounties;
+	} else {
+		var priorDate = new Date();
```
Comment (ebwinters) [inline utils/cache.js:24]: Can you do
```js
const priorDate = new Date().setDate(priorDate.getDate() - lastdays).toISOString().slice(0, 10);
```
It's cleaner and always prefer `const` over `var`.

File: utils/cache.js [hunk 2]
```diff
@@ -12,7 +12,27 @@ exports.currentStatus = {
...
+		priorDate.setDate(priorDate.getDate() - lastdays);
+		priorDate = priorDate.toISOString().slice(0, 10);
+		const priorData = [];
+		this.currentStatus.nytCounties.forEach((data) => {
+			if (data.date >= priorDate) {
+				priorData.push(data);
+			}
+		});
+		return priorData;
```
Comment (ebwinters) [inline utils/cache.js:33]: Can make this cleaner:
```js
return this.currentStatus.nytCounties.filter((data) => data.date >= priorDate));
```

File: utils/cache.js (hunk 1 above)
Comment (neo7337) [reply utils/cache.js:24]: I tried this but seems like priorDate is not declared yet so this `priorDate.getDate()` block won't work. So had to keep it the previous way and let it be var as we cannot assign values to a const.

File: tests/v3/covid-19/api_nyt.spec.js [hunk 3]
```diff
@@ -80,6 +80,15 @@ describe('TESTING /v3/covid-19/nyt/counties', () => {
 			});
 	});
 
+	it('/v3/covid-19/nyt/counties lastdays param', (done) => {
+		chai.request(app)
+			.get('/v3/covid-19/nyt/counties?lastdays=1')
+			.end((err, res) => {
+				testBasicProperties(err, res, 200, 'array');
+				done();
+			});
```
Comment (ebwinters) [inline tests/v3/covid-19/api_nyt.spec.js:89]: maybe also check if the length of the returned response is less than that of a normal counties length (probably in the 1000s), and also if it is greater than 1 just to make sure the param is working

**Comment by ebwinters** [comment]
@neo7337 I need you to do a couple things:
1. Add docs for this param
2. Add this param to other NYT endpoints, like /county
3. Fix the issue of checking lastDays=1 in prod. You may need to scrape this data more frequently by updating interval, or you may need to adjust to look Date-1-`lastdays` back.

**Pull request merged**